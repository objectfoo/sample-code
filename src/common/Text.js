import React from 'react';
import injectStyle from 'react-jss';
import classnames from 'classnames';

export const Text = ({
  component,
  type,
  classes,
  gutterBottom,
  gutterTop,
  className,
  children
}) => {
  const Component = component || componentFromType(type);

  return (
    <Component className={classnames(
      className,
      classes[type],
      { [classes.gutterBottom]: gutterBottom },
      { [classes.gutterTop]: gutterTop }
    )}>
      {children}
    </Component>
  );
};

const styles = theme => {
  return {
    body1: {
      margin: 0
    },
    title: {
      margin: 0
    },
    gutterBottom: {
      marginBotom: theme.spacing.unit
    },
    gutterTop: {
      marginTop: theme.spacing.unit
    }
  };
};

const StyledText = injectStyle(styles)(Text);
StyledText.displayName = 'StyledText';
export default StyledText;

const componentFromType = key => ({
  body1: 'p',
  title: 'h1'
}[key] || 'span');
