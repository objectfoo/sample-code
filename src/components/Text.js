import React from 'react';
import injectStyle from 'react-jss';
import classnames from 'classnames';

export const Text = ({
  component,
  type,
  classes,
  gutterBottom,
  gutterTop,
  children
}) => {
  const Component = component || componentFromType(type);

  return (
    <Component className={classnames(
      classes[type],
      { [classes.gutterBottom]: gutterBottom },
      { [classes.gutterTop]: gutterTop },
    )}>
      {children}
    </Component>
  )
};

const styles = theme => {
  return {
    body1: {
      margin: 0
    },
    title: {
      margin: 0
    },
    gutterBottom: {
      marginBotom: theme.spacing.unit
    },
    gutterTop: {
      marginTop: theme.spacing.unit
    }
  }
}

export default injectStyle(styles)(Text)

const componentFromType = key => ({
  body1: 'p',
  title: 'h1'
}[key] || 'span');
