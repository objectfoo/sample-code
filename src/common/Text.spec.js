import React from 'react';
import {shallow} from 'enzyme';
import {Text} from './Text';

describe('Text', () => {
  const mockClasses = {
    body1: 'body1',
    title: 'title',
    gutterBottom: 'gutterBottom',
    gutterTop: 'gutterTop'
  };

  it('renders text in a span when type not defined', () => {
    const expectedText = 'some text';
    const wrapper = shallow(<Text classes={mockClasses}>{expectedText}</Text>);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders gutter button', () => {
    const wrapper = shallow(<Text classes={mockClasses} gutterBottom />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders gutter top', () => {
    const wrapper = shallow(<Text classes={mockClasses} gutterTop />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders body1 with a paragraph', () => {
    const wrapper = shallow(<Text classes={mockClasses} type='body1' />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders h1 when type is title', () => {
    const wrapper = shallow(<Text classes={mockClasses} type='title' />);
    expect(wrapper).toMatchSnapshot();
  });

  it('allows override of default component when type not defined', () => {
    const wrapper = shallow(<Text classes={mockClasses} component='div' />);
    expect(wrapper).toMatchSnapshot();
  });

  it('allows override of default component when type is defined', () => {
    const wrapper = shallow(<Text classes={mockClasses} type='body1' component='div' />);
    expect(wrapper).toMatchSnapshot();
  });

  it('passes through className', () => {
    const expected = 'test';
    const wrapper = shallow(<Text classes={mockClasses} className={expected} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('merges className with default classes', () => {
    const expected = 'test';
    const expectedType = 'body1';

    const wrapper = shallow(
      <Text
        classes={mockClasses}
        type={expectedType}
        className={expected}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
