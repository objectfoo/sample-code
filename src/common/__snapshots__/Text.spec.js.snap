// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text allows override of default component when type is defined 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    classes={
        Object {
            "body1": "body1",
            "gutterBottom": "gutterBottom",
            "gutterTop": "gutterTop",
            "title": "title",
          }
    }
    component="div"
    type="body1"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "body1",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "body1",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Text allows override of default component when type not defined 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    classes={
        Object {
            "body1": "body1",
            "gutterBottom": "gutterBottom",
            "gutterTop": "gutterTop",
            "title": "title",
          }
    }
    component="div"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Text merges className with default classes 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    className="test"
    classes={
        Object {
            "body1": "body1",
            "gutterBottom": "gutterBottom",
            "gutterTop": "gutterTop",
            "title": "title",
          }
    }
    type="body1"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "test body1",
    },
    "ref": null,
    "rendered": null,
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "test body1",
      },
      "ref": null,
      "rendered": null,
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Text passes through className 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    className="test"
    classes={
        Object {
            "body1": "body1",
            "gutterBottom": "gutterBottom",
            "gutterTop": "gutterTop",
            "title": "title",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "test",
    },
    "ref": null,
    "rendered": null,
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "test",
      },
      "ref": null,
      "rendered": null,
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Text renders body1 with a paragraph 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    classes={
        Object {
            "body1": "body1",
            "gutterBottom": "gutterBottom",
            "gutterTop": "gutterTop",
            "title": "title",
          }
    }
    type="body1"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "body1",
    },
    "ref": null,
    "rendered": null,
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "body1",
      },
      "ref": null,
      "rendered": null,
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Text renders gutter button 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    classes={
        Object {
            "body1": "body1",
            "gutterBottom": "gutterBottom",
            "gutterTop": "gutterTop",
            "title": "title",
          }
    }
    gutterBottom={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "gutterBottom",
    },
    "ref": null,
    "rendered": null,
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "gutterBottom",
      },
      "ref": null,
      "rendered": null,
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Text renders gutter top 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    classes={
        Object {
            "body1": "body1",
            "gutterBottom": "gutterBottom",
            "gutterTop": "gutterTop",
            "title": "title",
          }
    }
    gutterTop={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "gutterTop",
    },
    "ref": null,
    "rendered": null,
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "gutterTop",
      },
      "ref": null,
      "rendered": null,
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Text renders h1 when type is title 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    classes={
        Object {
            "body1": "body1",
            "gutterBottom": "gutterBottom",
            "gutterTop": "gutterTop",
            "title": "title",
          }
    }
    type="title"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "title",
    },
    "ref": null,
    "rendered": null,
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "title",
      },
      "ref": null,
      "rendered": null,
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Text renders text in a span when type not defined 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    classes={
        Object {
            "body1": "body1",
            "gutterBottom": "gutterBottom",
            "gutterTop": "gutterTop",
            "title": "title",
          }
    }
>
    some text
</Unknown>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "some text",
      "className": "",
    },
    "ref": null,
    "rendered": "some text",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "some text",
        "className": "",
      },
      "ref": null,
      "rendered": "some text",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
